src/Sylius/Component/Core/Taxation/Applicator/OrderShipmentTaxesApplicator.php

```diff
@@ -59,8 +61,9 @@ public function __construct(
      */
     public function apply(OrderInterface $order, ZoneInterface $zone)
     {
+        /** @var TaxableInterface|ShipmentInterface $lastShipment */
         $lastShipment = $order->getLastShipment();
-        if (!$lastShipment) {
+        if (null === $lastShipment) {
```

Comment:
Actually we should use last shipment here! We should iterate through all shipments and apply taxes to all of them.